src/main/java/mops/foren/infrastructure/persistence/mapper/ThreadPageMapper.java

```diff
@@ -0,0 +1,32 @@
+package mops.foren.infrastructure.persistence.mapper;
+
+import mops.foren.domain.model.Thread;
+import mops.foren.domain.model.paging.Paging;
+import mops.foren.domain.model.paging.ThreadPage;
+import mops.foren.infrastructure.persistence.dtos.ThreadDTO;
+import org.springframework.data.domain.Page;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public abstract class ThreadPageMapper {
+
+    /**
+     * Map the Spring page to our page-type.
+     *
+     * @param dtoPage page of spring
+     * @param page    the number of the page
+     * @return Our PostPage object
+     */
+    public static ThreadPage toThreadPage(Page<ThreadDTO> dtoPage, int page) {
+        List<Thread> collect = dtoPage.stream()
```

Comment:
Perhaps a little bit unnecessary, but an improvement for readability ist dtoPage.getContent().stream()
or dtoPage.get().